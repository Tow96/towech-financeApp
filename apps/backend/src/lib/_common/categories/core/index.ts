export * from './category-aggregate';
export * from './category-events';
export * from './i-category-repository';
export * from './subcategory-entity';
export * from './subcategory-events';
