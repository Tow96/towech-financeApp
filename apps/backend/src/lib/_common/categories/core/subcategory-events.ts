// Slice packages
import { CategoryType } from './category-aggregate';

// Internal references
import { DomainEvent, DomainEventProps } from '../../primitives';

// ----------------------------------------------
export class SubCategoryCreatedEvent extends DomainEvent {
  readonly userId: string;
  readonly categoryId: string;
  readonly name: string;
  readonly type: CategoryType;

  constructor(props: DomainEventProps<SubCategoryCreatedEvent>) {
    super(props);
    this.userId = props.userId;
    this.categoryId = props.categoryId;
    this.name = props.name;
    this.type = props.type;
  }
}

export class SubCategoryUpdatedEvent extends DomainEvent {
  readonly categoryId: string;
  readonly type: CategoryType;
  readonly name: string;

  constructor(props: DomainEventProps<SubCategoryUpdatedEvent>) {
    super(props);
    this.categoryId = props.categoryId;
    this.type = props.type;
    this.name = props.name;
  }
}

export class SubCategoryRemovedEvent extends DomainEvent {
  readonly categoryId: string;

  constructor(props: DomainEventProps<SubCategoryRemovedEvent>) {
    super(props);
    this.categoryId = props.categoryId;
  }
}
