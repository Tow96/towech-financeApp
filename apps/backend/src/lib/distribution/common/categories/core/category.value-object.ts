// App packages
import { ValueObject } from '../../../../_common/primitives/value-object.base';

export enum CategoryType {
  income = 'INCOME',
  expense = 'EXPENSE',
  transfer = 'TRANSFER',
}

interface CategoryProps {
  _id: string;
  _userId: string;
  _type: CategoryType;
}

export class Category extends ValueObject<CategoryProps> {
  get id() {
    return this.props._id;
  }

  get userId() {
    return this.props._userId;
  }

  get type() {
    return this.props._type;
  }
}
