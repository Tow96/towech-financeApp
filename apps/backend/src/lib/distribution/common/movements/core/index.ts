export * from './category.value-object';
export * from './i-movement-repository';
export * from './movement-aggregate';
export * from './summary-item.value-object';
export * as events from './movement-events';
