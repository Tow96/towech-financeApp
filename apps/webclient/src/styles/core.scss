// The theming functions are heavily taken from angular material, but in a lighter way
// Original code: https://material.angular.io/
@use 'sass:map';

@function get-palette-hue($palette, $hue) {
  @return map.get($palette, $hue);
}

@function get-palette-hue-contrast($palette, $hue) {
  @return map.get(map.get($palette, contrast), $hue);
}
