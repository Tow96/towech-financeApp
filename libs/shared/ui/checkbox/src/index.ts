export * from './lib/shared-ui-checkbox.component';
