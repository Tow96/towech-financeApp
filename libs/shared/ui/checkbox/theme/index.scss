@use 'sass:map';
@use 'libs/shared/ui/themes' as core;

@mixin color($theme) {
  $colors: map.get($theme, color);
  $primary: map.get($colors, primary);
  $accent: map.get($colors, accent);
  $error: map.get($colors, error);

  towech-finance-checkbox .checkbox {
    input {
      background-color: core.get-palette-hue($primary, 600);
    }

    input:after {
      color: core.get-palette-hue($accent, 500);
    }

    input:focus-visible {
      box-shadow: 0 0 0 0.1rem rgba($color: core.get-palette-hue($accent, 300), $alpha: 0.5),
        inset rgba(0, 0, 0, 0.15) 1.95px 1.95px 2.6px !important;
    }

    input:focus + label {
      color: core.get-palette-hue($accent, 500);
    }
  }

  towech-finance-checkbox.ngrx-forms-invalid.ngrx-forms-dirty .checkbox {
    input:focus-visible {
      box-shadow: 0 0 0 0.1rem rgba($color: core.get-palette-hue($error, 300), $alpha: 0.5),
        inset rgba(0, 0, 0, 0.15) 1.95px 1.95px 2.6px !important;
    }

    input:after {
      color: core.get-palette-hue($error, 500);
    }

    input:focus + label {
      color: core.get-palette-hue($error, 500);
    }
  }
}
