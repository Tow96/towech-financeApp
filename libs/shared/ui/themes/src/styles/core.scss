// The theming functions are heavily taken from angular material, but in a lighter way
// Original code: https://material.angular.io/
@use 'sass:map';

@function get-palette-hue($palette, $hue) {
  @return map.get($palette, $hue);
}

@function get-palette-hue-contrast($palette, $hue) {
  @return map.get(map.get($palette, contrast), $hue);
}

@function define-theme(
  $background,
  $text,
  $primary,
  $accent,
  $success: null,
  $warning: null,
  $error: null
) {
  @if $background == null or $text == null {
    @error 'Background and text colors need to be defined';
  }

  @if $primary == null or $accent == null {
    @error 'primary and accent colors need to be defined palettes';
  }

  @if not map.get($primary, contrast) {
    @error 'Primary is not a palette';
  }

  @if not map.get($accent, contrast) {
    @error 'Accent is not a palette';
  }

  @if $success != null and not map.get($success, contrast) {
    @error 'Success is not a palette';
  }

  @if $error != null and not map.get($error, contrast) {
    @error 'Error is not a palette';
  }

  @if $warning != null and not map.get($warning, contrast) {
    @error 'Warning is not a palette';
  }

  $result: (
    color: (
      text: $text,
      background: $background,
      primary: $primary,
      accent: $accent,
      success: if($success != null, $success, $accent),
      warning: if($warning != null, $warning, $accent),
      error: if($error != null, $error, $accent),
    ),
  );

  @return $result;
}

@mixin color($theme) {
  $colors: map.get($theme, color);
  $primary: map.get($colors, primary);

  :root {
    background-color: map.get($colors, background);
    color: map.get($colors, text);
  }

  div.container {
    background-color: get-palette-hue($primary, 500);
  }
}
