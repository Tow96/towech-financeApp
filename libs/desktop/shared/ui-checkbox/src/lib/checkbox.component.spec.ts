// Libraries
import { ComponentFixture, TestBed } from '@angular/core/testing';
// Tested elements
import { DesktopCheckboxComponent } from './checkbox.component';
import { NG_VALUE_ACCESSOR } from '@angular/forms';

describe('Shared Input Component', () => {
  let component: DesktopCheckboxComponent;
  let fixture: ComponentFixture<DesktopCheckboxComponent>;
  let compiled: HTMLElement;
  let spy: jest.SpyInstance<any>;

  beforeEach(() => {
    jest.clearAllMocks();
    fixture = TestBed.createComponent(DesktopCheckboxComponent);
    component = fixture.componentInstance;
    // component.type = 'text';
    component.label = 'TEST';

    fixture.detectChanges();
    compiled = fixture.nativeElement;
  });

  it('Should be defined', () => expect(component).toBeTruthy());

  it('Must match the snapshot', () => expect(compiled).toMatchSnapshot());

  it('Should implement the NG_VALUE_ACCESSOR', () =>
    expect(fixture.debugElement.injector.get(NG_VALUE_ACCESSOR)).toBeTruthy());

  it('The changed and customTouched functions must do nothing when declared', () => {
    expect(component.customChanged(false)).toBeUndefined();
    expect(component.customTouched()).toBeUndefined();
  });

  describe('When writeValue is called', () => {
    it('Should update the value', () => {
      component.writeValue(true);
      expect(component.value).toBe(true);

      component.writeValue(false);
      expect(component.value).toBe(false);
    });
  });

  describe('When setOnChangeCallback is called', () => {
    it('Should update the changed function', () => {
      const newChanged = (value: boolean) => {
        console.log(value);
      };

      component.registerOnChange(newChanged);
      expect(component.customChanged).toBe(newChanged);
    });
  });

  describe('When setOnTouchedCallback is called', () => {
    it('Should update the touched function', () => {
      const newTouched = () => {
        console.log('NEW FUNCTION');
      };

      component.registerOnTouched(newTouched);
      expect(component.customTouched).toBe(newTouched);
    });
  });

  describe('When setIsDisabled is called', () => {
    it('Should set the disabled value', () => {
      component.setDisabledState(true);
      expect(component.disabled).toBe(true);
      component.setDisabledState(false);
      expect(component.disabled).toBe(false);
    });
  });

  describe('When on change is called', () => {
    let event: any;

    beforeEach(() => {
      event = { target: { checked: true } } as any;
      spy = jest.spyOn(component, 'customChanged');
      component.onChange(event);
    });
    it('Should call the customChanged function', () =>
      expect(spy).toHaveBeenCalledWith(event.target.checked));
  });

  describe('When the checkbox is clicked', () => {
    beforeEach(() => {
      spy = jest.spyOn(component, 'onChange');
      component.value = false;
      const checkbox = fixture.debugElement.nativeElement.querySelector('input');
      checkbox.click();
    });

    it('Should call the onChange function', () => expect(spy).toHaveBeenCalled());
  });

  describe('When the label is clicked', () => {
    beforeEach(() => {
      spy = jest.spyOn(component, 'onChange');
      component.value = false;
      const label = fixture.debugElement.nativeElement.querySelector('label');
      label.click();
    });

    it('Should call the onChange function', () => expect(spy).toHaveBeenCalled());
  });
});
