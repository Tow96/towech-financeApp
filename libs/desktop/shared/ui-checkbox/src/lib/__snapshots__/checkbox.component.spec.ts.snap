// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shared Input Component Must match the snapshot 1`] = `
<div
  id="root1"
>
  <div
    class="checkbox"
  >
    <input
      id="TEST"
      type="checkbox"
    />
    <label
      for="TEST"
    >
      TEST
    </label>
  </div>
</div>
`;
